Backport from master:
https://github.com/gpodder/gpodder/pull/1105

From 5f4af7d49c6735df9476939b3a71c5791c1834fd Mon Sep 17 00:00:00 2001
From: Oliver Smith <ollieparanoid@postmarketos.org>
Date: Fri, 23 Jul 2021 01:21:58 +0200
Subject: [PATCH 1/2] gtkui: fix loading of cached thumbnails

Due to this bug, thumbnails did not get loaded from the sqlite database,
but instead were regenerated on each start. Depending on the device this
leads to significant startup slowdown (30 podcast -> ~20s slowdown on my
PinePhone).
---
 src/gpodder/gtkui/model.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gpodder/gtkui/model.py b/src/gpodder/gtkui/model.py
index a31e1075..40eedf6c 100644
--- a/src/gpodder/gtkui/model.py
+++ b/src/gpodder/gtkui/model.py
@@ -726,6 +726,7 @@ class PodcastListModel(Gtk.ListStore):
             if self._max_image_side not in (pixbuf.get_width(), pixbuf.get_height()):
                 logger.debug("cached thumb wrong size: %r != %i", (pixbuf.get_width(), pixbuf.get_height()), self._max_image_side)
                 return None
+            return pixbuf
         except Exception as e:
             logger.warn('Could not load cached cover art for %s', channel.url, exc_info=True)
             channel.cover_thumb = None
-- 
2.20.1

